main/midnight/src/ui/uielement.h

```diff
@@ -6,6 +6,18 @@
 //
 //
 
+// Windows build needs definition upfront
```

Comment:
Ugly but was force to allow the typedef definitions outside the 'ifndef' nlock. VS would not find them otherwise. Wrapped around _OS_WINDOWS_BUILD_ to not break other platform builds.

Reply:
Is the position fixed here? This works pretty much anywhere on Mac. So it doesn't need to be wrapped in _OS_WINDOWS_BUILD_ 

This would be ideal if it builds on Windows unless it's the  mxtypes.h which breaks it, in which case it could go between the two. 

```
    #ifndef __UIELEMENT_H_INCLUDED__
    #define __UIELEMENT_H_INCLUDED__

    #include "../cocos.h"
    #include "../library/inc/mxtypes.h"

    // Windows build needs definition upfront
    namespace chilli {
        namespace ui {

            typedef cocos2d::ui::AbstractCheckButton::ccWidgetClickCallback WidgetClickCallback;
            typedef cocos2d::ui::AbstractCheckButton::ccWidgetEventCallback WidgetEventCallback;
        }
    }

    FORWARD_REFERENCE(uieventargs);
    FORWARD_REFERENCE(uielement);
    FORWARD_REFERENCE(uinotificationinterface);
```

Reply:
It's not the order that matters. What seems to be happening is that the '#ifndef' declaration at top of .h file is making the typedef definition not be visible in uihelper.h 'WidgetEventCallback' use.  If this was a class definition I could use forward reference for class in 'uihelper.h' but since this is a typedef I cannot. Since this is a typedef declaration it *should* be fine to define it outside the #ifndef block for all platforms (if you're ok with it).

Bottom line: typedef is not visible in uihelper.h when inside the #ifndef (circularity present somehwere?).